storage/database/db_manager.go

```diff
@@ -1057,30 +1153,106 @@ func (dbm *databaseManager) WriteMerkleProof(key, value []byte) {
 
 // Cached Trie Node operation.
 func (dbm *databaseManager) ReadCachedTrieNode(hash common.Hash) ([]byte, error) {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(hash[:]); err == nil {
+			return val, nil
+		}
+	}
+	return dbm.ReadCachedTrieNodeFromOld(hash)
+}
+
+// Cached Trie Node Preimage operation.
+func (dbm *databaseManager) ReadCachedTrieNodePreimage(secureKey []byte) ([]byte, error) {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(secureKey); err == nil {
+			return val, nil
+		}
+	}
+	return dbm.ReadCachedTrieNodePreimageFromOld(secureKey)
+}
+
+// State Trie Related operations.
+func (dbm *databaseManager) ReadStateTrieNode(key []byte) ([]byte, error) {
+	if dbm.inMigration {
+		if val, err := dbm.newStateTrieDB.Get(key); err == nil {
+			return val, nil
+		}
+	}
+	return dbm.ReadStateTrieNodeFromOld(key)
```

Comment:
How about use `HasStateTrieNodeFromOld` instead. The processing for the value returned from the DB has disappeared. At `line 1078` in the `HasStateTrieNode` method.